gcloud/credentials.py

```diff
@@ -59,6 +68,113 @@ def get_credentials():
     return client.GoogleCredentials.get_application_default()
 
 
+def _store_user_credential(credential):
+    """Stores a user credential as a well-known file.
+
+    Prompts user first if they want to store the minted token and
+    then prompts the user for a filename to store the token
+    information in the format needed for get_credentials().
+
+    :type credential: :class:`oauth2client.client.OAuth2Credentials`
+    :param credential: A user credential to be stored.
+    """
+    ans = six.moves.input('Would you like to store your tokens '
+                          'for future use? [y/n] ')
+    if ans.strip().lower() != 'y':
+        return
+
+    filename = six.moves.input('Please name the file where you wish '
+                               'to store them: ').strip()
+
+    payload = {
+        'client_id': credential.client_id,
+        'client_secret': credential.client_secret,
+        'refresh_token': credential.refresh_token,
+        'type': 'authorized_user',
+    }
+    with open(filename, 'w') as file_obj:
+        json.dump(payload, file_obj, indent=2, sort_keys=True,
+                  separators=(',', ': '))
+        file_obj.write('\n')
+
+    print 'Saved %s' % (filename,)
+    print 'If you would like to use these credentials in the future'
+    print 'without having to initiate the authentication flow in your'
+    print 'browser, please set the GOOGLE_APPLICATION_CREDENTIALS'
+    print 'environment variable:'
+    print '    export GOOGLE_APPLICATION_CREDENTIALS=%r' % (filename,)
+    print 'Once you\'ve done this, you can use the get_credentials()'
+    print 'function, which relies on that environment variable.'
+    print ''
+    print 'Keep in mind, the refresh token can only be used with the'
+    print 'scopes you granted in the original authorization.'
```

Comment:
ISTM that anything using `print` or `input` should be in a function registered as a setuptools `console_scripts` entry point, rather than happening as a side-effect of a library call.